cmd/commands/checksum.go

```diff
@@ -0,0 +1,58 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/* Copyright Contributors to the cpackget project. */
+
+package commands
+
+import (
+	"github.com/open-cmsis-pack/cpackget/cmd/cryptography"
+	errs "github.com/open-cmsis-pack/cpackget/cmd/errors"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+var checksumCreateCmdFlags struct {
+	// hashAlgorithm is the cryptographic hash function to be used
+	hashAlgorithm string
+
+	// outputDir is the target directory where the checksum file is written to
+	outputDir string
+}
+
+func init() {
+	ChecksumCreateCmd.Flags().StringVarP(&checksumCreateCmdFlags.hashAlgorithm, "hash-algorithm", "a", "", "specifies the hash function to be used")
+	ChecksumCreateCmd.Flags().StringVarP(&checksumCreateCmdFlags.outputDir, "output-dir", "o", "", "specifies output directory for the checksum file")
+}
+
+var ChecksumCreateCmd = &cobra.Command{
+	Use:   "checksum-create [<local .path pack>]",
+	Short: "Generates a .checksum file containing the digests of a pack",
+	// TODO: Long, show valid hash algorithms
+	Args:              cobra.MinimumNArgs(0),
```

Comment:
A few lines later, you're checking 

```golang
if len(args) == 0 {
```
You could simply avoid the `if` by changing the minimum number of args to 
```golang
Args:              cobra.MinimumNArgs(1),
```

Then cobra will validate the number of args automatically.

Reply:
That one is a bit different because args can actually be empty and the pack names will be read from a file specified by `-f/--packs-list-filename` and later be appended to the cmd args: https://github.com/Open-CMSIS-Pack/cpackget/blob/85b352851370963377cc860c3c89f09cd42fcfd4/cmd/commands/add.go#L72. So in that case, if I have min-num-of-args set to 1, it'll always require a pack in args even if you're specifying packs via a file.

Reply:
Done!